File: Java/프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(long num) {
        /*
        1. 1이 될 때까지 
        2. 짝수 : 2로 나누기 , count++
        3. 홀수 : 3을 곱하고 1을 더하기 , count++
        4. num이 1이면 0을 리턴 
        5. count가 500번 될 때까지 1이 되지 않으면 -1을 리턴
        */
        int count = 0; 
        
        while(num > 1){
            if(num % 2 == 0){
                num = num / 2;
            }
            else{
                num = (num * 3) + 1; 
            }  
            count++;
            
            if(count > 500){
                return -1;
            }
        }
        
        return count;
    }
}

File: Python/프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    def dfs(number, cnt):
        if number == 1:
            return cnt
        if cnt == 500:
            return -1
        
        if number % 2 == 0:
            return dfs(number//2, cnt+1)
        else:
            return dfs(number*3+1, cnt+1)
    return dfs(num, 0)